pylint/checkers/variables.py

```diff
@@ -1342,6 +1337,9 @@ def leave_module(self, node: nodes.Module) -> None:
 
     def visit_classdef(self, node: nodes.ClassDef) -> None:
         """Visit class: update consumption analysis variable."""
+        for name in node.nodes_of_class(nodes.AssignName):
+            # Names have to be inferred, because they could affect locals.
+            safe_infer(name)
```

Comment:
This is quite lame, but we'll need PyCQA/astroid#1490 to get rid of it.

Reply:
Worked around this.

Reply:
... but will still need an astroid update to pass the test suite. PR forthcoming.

Reply:
See https://github.com/pylint-dev/astroid/pull/2119

Reply:
Should we backport the astroid update (even if this is used in a false negative that we won't backport) ?

Reply:
Good question, but I'd rather not. It's not a very stable change.